lib/stat_tracker.rb

```diff
@@ -65,8 +56,35 @@ def total_tied_games
    end.size) / 2
   end
 
-# ==================       Game Stats Methods      ==================
+  def total_goals_per_game
+    @games.reduce({}) do |ids_to_scores, game|
+      ids_to_scores[game.game_id] = game.away_goals + game.home_goals
+      ids_to_scores
+    end
+  end
+
+  def total_games_per_team(exclude_hoa = nil)
+    @game_teams.reduce(Hash.new(0)) do |result, game_team|
+      result[game_team.team_id] += 1 unless game_team.hoa == exclude_hoa
+      result
+    end
+  end
 
+  def total_goals_per_team(exclude_hoa = nil)
+    @game_teams.reduce(Hash.new(0)) do |result, game_team|
+      result[game_team.team_id] += game_team.goals unless game_team.hoa == exclude_hoa
+      result
+    end
+  end
+
+  def average_goals_per_game_per_team(exclude_hoa = nil)
+    @teams.reduce(Hash.new(0)) do |result, team|
+      average = (total_goals_per_team(exclude_hoa)[team.id] / total_games_per_team(exclude_hoa)[team.id].to_f).round(2)
+      result[team] = average unless average.nan?
```

Comment:
This is great. What does .nan? mean?

Reply:
I think it stands for Not A Number! I had to do some googling to get that working. It was giving me an error because our fixture files are missing data so sometimes it divides by zero.

Reply:
That's clever! I've never seen that before!